capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -47,37 +47,11 @@ public ClubServlet() throws IOException {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Club> result = new ArrayList<>();
-    if (request.getParameter("id") != null) {
-      DocumentReference docRef = clubs.document(request.getParameter("id"));
-      ApiFuture<DocumentSnapshot> future = docRef.get();
-      DocumentSnapshot document = null;
-      Club club = null;
-      try {
-        document = future.get();
-        if (document.exists()) {
-          club = document.toObject(Club.class);
-        } else {
-          System.err.println("Error: no such document!");
-        }
-      } catch (Exception e) {
-        System.err.println("Error: " + e);
-      }
-      result.add(club);
+    List<Club> retrievedClubs = Utility.get(clubs, request, response, new GenericClass(Club.class));
+    if (retrievedClubs != null) {
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(retrievedClubs));
```

Comment:
Much cleaner get method!

Reply:
Thank you! Should be super easy for all our servlets to use this in the future.